esp-hal/src/time.rs

```diff
@@ -50,9 +50,10 @@ pub fn current_time() -> fugit::Instant<u64, 1, 1_000_000> {
 
 #[cfg(esp32)]
 pub(crate) fn time_init() {
+    let apb = Clocks::get().apb_clock.to_Hz();
     // we assume 80MHz APB clock source - there is no way to configure it in a
     // different way currently
-    const APB_FREQUENCY: u32 = 80_000_000u32;
+    assert!(apb, 80_000_000u32);
```

Comment:
I figured it was a good idea to check this assumption rather than silently break on new chips, and given that this function is only called once we don't have to worry about perf.

Reply:
We don't have Clocks here initialised yet.

Reply:
Hmm, maybe we should move this init to `esp_hal::init`, now that we have that :thinking:. Then clocks will be initialized.

Reply:
We can move a lot of things in there now :)